service/storage_proxy.cc

```diff
@@ -2012,7 +2020,7 @@ storage_proxy::create_write_response_handler(const std::tuple<lw_shared_ptr<paxo
                     inet_address_vector_topology_change(), inet_address_vector_topology_change(), std::move(tr_state), get_stats(), std::move(permit));
 }
 
-void storage_proxy::register_cdc_operation_result_tracker(const std::vector<storage_proxy::unique_response_handler>& ids, lw_shared_ptr<cdc::operation_result_tracker> tracker) {
+void storage_proxy::register_cdc_operation_result_tracker(const utils::small_vector<storage_proxy::unique_response_handler, 3>& ids, lw_shared_ptr<cdc::operation_result_tracker> tracker) {
```

Comment:
Why do you optimize for a batch with up to three different statements? One should be a common case here.

Reply:
I thought we have one per replica. Agree it should be 1 if it's per-mutation.